src/utils.ts

```diff
@@ -0,0 +1,138 @@
+import { SchemaProperty } from './components/form-element'
+
+export interface Conditional {
+    compiled: string
+    then?: SchemaProperty
+    else?: SchemaProperty
+}
+
+//checks if item is object
+export const isObject = (item: any) => {
+    return item && typeof item === 'object' && !Array.isArray(item)
+}
+
+//merges the current object at all nesting levels with a new one
+//the method does not return a new object; the current one is being altered
+export const addProperties = (currentObject: any, newProperties: any): any => {
+    if (isObject(currentObject) && isObject(newProperties)) {
+        for (const key in newProperties) {
+            if (isObject(newProperties[key])) {
+                if (!currentObject[key]) {
+                    Object.assign(currentObject, { [key]: {} })
+                }
+                addProperties(currentObject[key], newProperties[key])
+            } else {
+                if(key === 'required' && Array.isArray(newProperties[key]) && Array.isArray(currentObject[key])){
```

Comment:
hmmm ... is there a chance that the Schema could be defined with uppercase key names? "REQUIRED": "whatever" ? If yes I hate the one who does this!

Reply:
hate ones that write using uppercase

Reply:
this case will be ignored for now :)